dvc/remote/base.py

```diff
@@ -745,8 +754,11 @@ def gc(self, named_cache, jobs=None):
             path_info = self.checksum_to_path_info(checksum)
             if self.is_dir_checksum(checksum):
                 self._remove_unpacked_dir(checksum)
+            self.index.remove(checksum)
```

Comment:
Wonder how fast this is, esp when we are running this for each hash. Maybe worth just dropping index and reindexing the next time?